### PR TITLE
Refactor ui.queryArguments state and introduce a new reducer for the route

### PR DESCRIPTION
**Pull request by Tug**
Creates new reducer `ui.route` which combines the two reducers `ui.route.query` and `ui.route.path` each combining the initial and current query arguments and path of the application route.

For instance if the app loaded from `/log-in?retry=1` and redirects to `/log-in`, the `ui.route` state will be:
```js
{
    query: {
        initial: { retry: "1" }
        current: {},
    },
    path: {
        initial: '/log-in',
        current: '/log-in',
    }
}
```

This change was initially introduced in https://github.com/Automattic/wp-calypso/pull/19036 so we group path and query arguments in the same reducer.

**Important**: With this patch, we also change the default caching behaviour of the `ui` state so we may persist that information for SSR.

### Testing Instructions
- Boot this branch locally

- Try accessing a few routes with arguments and for each one check that the sate is consistant:

- http://calypso.localhost:3000/log-in?redirect_to=https%3A%2F%2Fwordpress.com%2Fthemes
State should look like:
```
{
    route: {
        path: {
            initial: '/log-in',
            current: '/log-in'
        },
        query: {
            initial: {
                redirect_to: 'https://wordpress.com/themes',
                _timestamp: ...
            },
            current: {
                redirect_to: 'https://wordpress.com/themes',
                _timestamp: ...
            }
      }
}
```

- http://calypso.localhost:3000/log-in?client_id=50916&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D50916
(or another tour you haven't done yet)
State should look like: 
```
{
    route: {
        path: {
            initial: '/log-in',
            current: '/log-in'
        },
        query: {
            initial: {
              client_id: '50916',
              redirect_to: 'https://public-api.wordpress.com/oauth2/authorize?response_type=code&client_id=50916',
              _timestamp: ...
            },
            current: {
                client_id: '50916',
                redirect_to: 'https://public-api.wordpress.com/oauth2/authorize?response_type=code&client_id=50916',
                _timestamp: ...
            }
        }
    }
}
```
- Check that tours can boot by adding `?tour=main` at the end of your url or any other tour other than `main` that you did not try yet. Some tours seems to be broken, so checking that you can complete them should not be part of this PR review.

### Reviews
- [ ] Product
- [ ] Tests
- [ ] Code

### CODE REVIEW

File: client/state/selectors/get-current-query-arguments.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
```
Comment (stephanethomas) [inline client/state/selectors/get-current-query-arguments.js]: There is an unnecessary blank lines here.

Same thing in the other selectors.

File: client/state/selectors/get-current-query-arguments.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ *
+ * @format
```
Comment (stephanethomas) [inline client/state/selectors/get-current-query-arguments.js]: My understanding is that `@format` should be on its own line.

Same thing in the other selectors.

File: client/state/selectors/get-current-query-arguments.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ */
+
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
```
Comment (stephanethomas) [inline client/state/selectors/get-current-query-arguments.js]: This comment block can be removed.

Same thing in the other selectors.

File: client/state/ui/route/path/reducer.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
```
Comment (stephanethomas) [inline client/state/ui/route/path/reducer.js]: We don't need that comment block.

File: client/state/ui/route/reducer.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
```
Comment (stephanethomas) [inline client/state/ui/route/reducer.js]: We don't need that comment block.

File: client/state/selectors/get-current-query-arguments.js (hunk 2 above)
Comment (Tug) [reply client/state/selectors/get-current-query-arguments.js]: I believe the original codemod added it to the first comment, then another codemods replaced that https://github.com/Automattic/wp-calypso/pull/19711

I don't mind using the new syntax 👍

File: client/state/ui/route/path/reducer.js (hunk 4 above)
Comment (Tug) [reply client/state/ui/route/path/reducer.js]: Yes we do, the linter will complain otherwise...

**Pull request merged**